lib/publish.js

```diff
@@ -43,12 +43,16 @@ class Publish extends BaseCommand {
 
     log.verbose('publish', args)
 
-    const opts = { ...this.npm.flatOptions }
-    const { unicode, dryRun, json, defaultTag } = opts
+    const unicode = this.npm.config.get('unicode')
+    const dryRun = this.npm.config.get('dry-run')
+    const json = this.npm.config.get('json')
+    const defaultTag = this.npm.config.get('tag')
 
     if (semver.validRange(defaultTag))
       throw new Error('Tag name must not be a valid SemVer range: ' + defaultTag.trim())
 
+    const opts = { ...this.npm.flatOptions }
```

Comment:
note: the flattening of `publishConfig` will be munged by subsequent PRs for great justice.